File: CS/WinForms.Client/DTO.cs
namespace WinForms.Client {
    public class DTO {
        public Guid ID { get; set; }
        public string ProductName { get; set; }
        public decimal UnitPrice { get; set; }
        public int Quantity { get; set; }
        public float Discount { get; set; }
        public DateTime OrderDate { get; set; }
        //
        public static DTO Create(OData.Server.BusinessObjects.OrderItem item) {
            return new DTO() {
                ID = item.ID,
                ProductName = item.ProductName,
                UnitPrice = item.UnitPrice,
                Quantity = item.Quantity,
                Discount = item.Discount,
                OrderDate = item.OrderDate.Date
            };
        }
        public void ApplyChanges(OData.Server.BusinessObjects.OrderItem item) {
            item.ID = ID;
            item.OrderDate = OrderDate;
            item.Quantity = Quantity;
            item.Discount = Discount;
            item.UnitPrice = UnitPrice;
            item.ProductName = ProductName;
        }
    }

}

File: VB/WinForms.Client/DTO.vb
Namespace WinForms.Client

    Public Class DTO

        Public Property ID As Guid

        Public Property ProductName As String

        Public Property UnitPrice As Decimal

        Public Property Quantity As Integer

        Public Property Discount As Single

        Public Property OrderDate As DateTime

        '
        Public Shared Function Create(ByVal item As OData.Server.BusinessObjects.OrderItem) As DTO
            Return New DTO() With {item.ID, item.ProductName, item.UnitPrice, item.Quantity, item.Discount, .OrderDate = item.OrderDate.[Date]}
        End Function

        Public Sub ApplyChanges(ByVal item As OData.Server.BusinessObjects.OrderItem)
            item.ID = ID
            item.OrderDate = OrderDate
            item.Quantity = Quantity
            item.Discount = Discount
            item.UnitPrice = UnitPrice
            item.ProductName = ProductName
        End Sub
    End Class
End Namespace
